assign2/index.js

```diff
@@ -0,0 +1,302 @@
+import { HISTORY_LIST } from "./DATA.js";
+
+// 1.b
+const list = document.querySelector("#list");
+
+function makeList(MoneyList, container) {
+  MoneyList.forEach((data, index) => {
+    const { title, summary, money, getto } = data;
+
+    const listWrapper = document.createElement("article");
+    listWrapper.classList.add("list_wrapper");
+    listWrapper.classList.add(
+      getto ? "plus_list_wrapper" : "minus_list_wrapper"
+    );
+
+    const h5 = document.createElement("h5");
+    h5.innerText = title;
+
+    const h4 = document.createElement("h4");
+    h4.innerText = summary;
+
+    const h3 = document.createElement("h3");
+    h3.innerText = getto
+      ? money.toLocaleString()
+      : "-" + money.toLocaleString();
+    h3.classList.add(getto ? "plus" : "minus");
+
+    const div = document.createElement("div");
+    div.classList.add("money_and_btn");
+    const button = document.createElement("button");
+    button.classList.add(index);
+    listWrapper.classList.add(index);
+    button.innerText = "x";
+    div.appendChild(button);
+    div.appendChild(h3);
+
+    listWrapper.append(h5, h4, div);
+    container.appendChild(listWrapper);
+  });
+}
+
+makeList(HISTORY_LIST, list);
+
+//나의 자산 반영
+let INIT_BALANCE = 0;
+const totalMoney = document.querySelector("#my_money > h3");
+
+function myBalance(MoneyList) {
+  MoneyList.forEach((data) => {
+    const { money, getto } = data;
+    INIT_BALANCE += getto ? money : -money;
+    totalMoney.innerText = INIT_BALANCE.toLocaleString();
+  });
+}
+
+myBalance(HISTORY_LIST);
+
+//2.
+let IN_MONEY = 0;
+let OUT_MONEY = 0;
+const plusMoney = document.getElementById("plus_money");
+const minusMoney = document.getElementById("minus_money");
+
+HISTORY_LIST.forEach((element) => {
+  const { money, getto } = element;
+  getto ? (IN_MONEY += money) : (OUT_MONEY += money);
+  plusMoney.innerText = IN_MONEY.toLocaleString();
+  minusMoney.innerText = OUT_MONEY.toLocaleString();
+});
+
+//3.
+//버튼 불러오기
+const inputPlus = document.getElementById("input_plus");
+const inputMinus = document.getElementById("input_minus");
+const HIDDEN_CLASS = "hidden";
+
+//리스트 필터
+function filterList() {
+  const plusListWrapper = document.querySelectorAll(".plus_list_wrapper");
+  const minusListWrapper = document.querySelectorAll(".minus_list_wrapper");
+
+  if (!inputPlus.checked) {
+    hiddenList(plusListWrapper);
+  } else if (inputPlus.checked) {
+    showList(plusListWrapper);
+  }
+
+  if (inputMinus.checked) {
+    showList(minusListWrapper);
+  } else {
+    hiddenList(minusListWrapper);
+  }
+}
+
+//리스트 보여주기
+function showList(listWrapper) {
+  listWrapper.forEach((element) => {
+    element.classList.remove(HIDDEN_CLASS);
+  });
+}
+
+//리스트 숨기기
+function hiddenList(listWrapper) {
+  listWrapper.forEach((element) => {
+    element.classList.add(HIDDEN_CLASS);
+  });
+}
+
+//event 붙이기
+inputPlus.addEventListener("change", () => filterList());
+inputMinus.addEventListener("change", () => filterList());
```

Comment:
```suggestion
inputPlus.addEventListener("change", filterList);
inputMinus.addEventListener("change", filterList);
```

혹시 화살표 함수를 써준 다른 이유가 있을까?! 매개변수로 넘겨줘야 하는게 없는데도 저렇게 쓴 이유가 궁금해!

Reply:
정신없음이슈,,,ㅎ